src/comm.cpp

```diff
@@ -448,19 +448,20 @@ void game_loop_unix(int control) {
             /* Waitstate the character */
             if (d->character && d->character->wait)
                 continue;
-            read_from_buffer(d);
-            if (d->incomm[0] != '\0') {
+
+            if (auto incomm = d->pop_incomm(); incomm) {
```

Comment:
std::optional's syntax in an if is kind of weird but makes sense :)

Reply:
Nice to see that you were able to ditch the copying here. It seemed...unnatural.

Reply:
Yay! Thanks :) In fact the `; incomm)` is unnecessary; the `if()` doesn't need it in this case (it was refactored from when it used to be `...; !incomm.empty())`